### DISCUSSION TITLE
How does CI/CD work with Elsa?

### DISCUSSION THREAD

**Discussion by cohenoshriconsulting** [elsa-workflows/elsa-core#2996] [Design & tech]
My one fear changes to a critical workflow in production being out of sync with our deployment.
Is there a way to design the workflow and save a JSON to be deployed at a later time through our pipeline?
Maybe even have the JSON stored in our git and pulled automatically on changes>

**Comment by sfmskywalker** [comment]
There are various ways to handle this, but the two most common ones are:

1. Use the designer to create the workflow and export it to a JSON file. You can then store this file in some folder in your repository and have your application read workflow files from this directory. [This sample project](https://github.com/elsa-workflows/elsa-core/tree/master/src/samples/aspnet/Elsa.Samples.SignalApi) uses this technique.
2. Create your workflows using the fluent builder API. For small workflows this could work well, and you can still visualize them via the designer.
